### ISSUE TITLE
SimCLR example

### ISSUE THREAD

**Issue by sdesrozis** [pytorch/ignite#1219]
## 🚀 Feature

Idea is to provide new class of example for self supervised learning.

It could be an implementation of paper "A Simple Framework for Contrastive Learning of Visual Representations" https://arxiv.org/abs/2002.05709

Things to do are
* normalized temperature-scaled cross entropy loss (NT-Xent loss)
* Pair-transformed dataset
* CIFAR example using resnet
* compare to imagnet pretraining
* use idist for distributed training

**Comment by vfdev-5** [comment]
This also may be a good FR for `ignite.data` module.

**Comment by vfdev-5** [comment]
Maybe, it would be better to reimplement BYOL too : https://arxiv.org/abs/2006.07733

**Comment by sdesrozis** [comment]
Yes, it's not so far from SimCLR. I will do it in my research lab during next few months.
